nav2_behavior_tree/plugins/decorator/is_new_path_shorter.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2022 Pradheep Padmanabhan - Neobotix GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <memory>
+#include <vector>
+#include "nav2_util/geometry_utils.hpp"
+
+#include "nav2_behavior_tree/plugins/decorator/is_new_path_shorter.hpp"
+
+namespace nav2_behavior_tree
+{
+
+IsNewPathShorter::IsNewPathShorter(
+  const std::string & name,
+  const BT::NodeConfiguration & conf)
+: BT::RetryNode(name, conf)
+{
+  node_ = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
+
+  // Initializing the paths
+  new_path.poses.resize(1);
+  new_path.poses[0].pose.position.x = 0.0;
+  old_path.poses.resize(1);
+  old_path.poses[0].pose.position.x = 0.0;
+}
+
+inline BT::NodeStatus IsNewPathShorter::tick()
+{
+  getInput("path", new_path);
+  getInput("allowed_goal_distance_proximity", allowed_goal_distance_proximity);
+
+  // Check if the old path is not same with the current one
+  if (new_path != old_path && old_path.poses.size() != 0 && new_path.poses.size() != 0) {
+    // Calculate and compare the old and the new path length, given the goal proximity
+    if ( (nav2_util::geometry_utils::calculate_path_length(new_path, 0) -
```

Comment:
well, the idea was to `if new computed path is significantly longer than current path & close to goal, then pause, and replan goal after waiting`. So that is the reason why I subtract the path lengths, and use a distance(allowed_goal_distance_proximity), to check if the current path is significantly shorter or not. 

Maybe I misunderstood something? or again it's just the name of the variable causing the issue?

Reply:
Got it, I think there's 2 variables required
- Distance from goal to apply this behavior at all , basically a boolean for how far we are from the goal to even allow this behavior to take place
- Change in distance that would trigger this behavior, if "significantly" longer distance measure

Reply:
There's also the issue to consider about replanning infrequently. There needs to be a check here relative to a fair starting point in case we only replan every 10 seconds or something. Then the actual robot's location would be further from the start of the path than would be a reasonable comparison.

Reply:
I indeed vouch for the idea of having 2 variables, bringing in much more flexibility to the feature.

> IsPathLongerOnApproach (invert the logic) then would be clear that if its longer on approach, then pause and wait, if the path is a multiple longer and within N meters of the goal.

This idea clearly expresses the idea of how the BT node should look like.